pkg/systemd/services/service.jsx

```diff
@@ -21,50 +21,130 @@ import React from "react";
 import { ServiceDetails } from "./service-details.jsx";
 import { LogsPanel } from "cockpit-components-logs-panel.jsx";
 import { superuser } from 'superuser';
 import { WithDialogs } from "dialogs.jsx";
 
+import s_bus from "./busnames.js";
+
 import cockpit from "cockpit";
 
 const _ = cockpit.gettext;
 
+function debug() {
+    if (window.debugging == "all" || window.debugging?.includes("service-details"))
+        console.debug.apply(console, arguments);
+}
+
 export class Service extends React.Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            /* The initial load of the Services page will not call GetAll for units Properties
-             * since ListUnits API call already has provided us with a subset of the Properties.
-             * As a result, properties like the 'Requires' are not present in the state at this point.
-             * If it's the first time to open this service's details page we need to fetch
-             * the unit properties by calling getUnitByPath.
-             */
-            shouldFetchProps: props.unit.Names === undefined,
+            unit_id: null,
+            unit_props: null,
+            error: null,
+            reloading: false,
         };
+
+        this.path = null;
+        this.subscriptions = [];
+    }
+
+    async componentDidMount() {
+        const dbus = this.props.dbusClient;
+        const [path] = await dbus.call(s_bus.O_MANAGER, s_bus.I_MANAGER, "LoadUnit", [this.props.unitId]);
+        this.path = path;
+
+        this.subscriptions.push(dbus.subscribe(
+            { interface: s_bus.I_MANAGER, member: "Reloading" },
+            (_path, _iface, _signal, [reloading]) => {
+                debug("Service detail", this.props.unitId, "Reloading", reloading);
+                this.setState({ reloading });
+                if (!reloading)
+                    this.updateProperties();
+            }));
+
+        this.subscriptions.push(dbus.subscribe(
+            { path, interface: s_bus.I_PROPS, member: "PropertiesChanged" }, () => {
+                debug("Service detail", this.props.unitId, "PropertiesChanged");
+                this.updateProperties();
+            }));
+
+        /* We need this *only* to pick up failed Conditions after attempting to start a service, as the unit immediately gets
+         * unloaded again and we don't get a PropertiesChanged signal. */
+        this.subscriptions.push(dbus.subscribe(
+            { interface: s_bus.I_MANAGER, member: "JobRemoved" }, (_p, _i, _s, [_job_id, _job_path, unit, result]) => {
+                if (unit === this.props.unitId && result === "done") {
+                    debug("Service detail", this.props.unitId, "JobRemoved");
+                    this.updateProperties();
+                }
+            }));
+
+        debug("Service detail", this.props.unitId, "componentDidMount, loading initial properties");
+        this.updateProperties();
```

Comment:
`await`?  (Just because updateProperties is declared `async`.  I guess it doesn't matter, but it is cleaner IMO.)

Reply:
Good point, yes. I marked all event handler arrows as `async` too, and added the awaits. Committed locally, and passes local test runs. But I won't push that by itself, due to scarce CI. Thanks for spotting!